frontend/static/js/spongebob.js

```diff
@@ -614,6 +614,10 @@ function trackForm(form) {
 		console.log("Form changed!");
 		checkForUnsavedChanges();
 	})
+	$(`[form="${form.id}"]`).change(function() {
```

Comment:
This should be in `yagInitUnsavedForms` in spongebob.js, all form tracking starts from there.

Reply:
I'm happy to move it there if you feel it's more appropriate, but I placed this code in `trackForm` (which is called by `yagInitUnsavedForms` anyway) since the main listener is attached there as well. The full context is
```js
// in trackForm...

	$(form).change(function () {
		console.log("Form changed!");
		checkForUnsavedChanges();
	}) // code before this change
	$(`[form="${form.id}"]`).change(function() {
		console.log("Form changed!");
		checkForUnsavedChanges()
	}) // ADDED
```

Reply:
I mean you could just append the element with that selector to the list of forms in `yagInitUnsavedForms` and not have redundant change handler for it again here.

Reply:
I am not sure at a glance how we can make that work easily. `trackForm` needs to know what the main form element is, so we cannot treat `form`s and elements with `[form="id"]` the same way.

Reply:
I was thinking something like this, additionally your current logic doesn't  ignores tracking changes on the element with [form="id"] if it doesn't have the `no-unsaved-popup` class. 
```diff
 function yagInitUnsavedForms(selectorPrefix) {
        let unsavedForms = $(selectorPrefix + "form")
        unsavedForms.each(function (i, rawElem) {
-              if (rawElem.classList.contains("no-unsaved-popup")) return;
               trackForm(rawElem);
+             $(`[form="${rawElem.id}"]`).each((i,e) => trackForm(e) )
        });
 }
 
 function trackForm(form) {
+  if (form.classList.contains("no-unsaved-popup")) return;
        let savedVersion = serializeForm($(form));
 
        let hasUnsavedChanges = false
```

Reply:
oh wait, the "[form="${rawElem.id}"] is not necessarily a form, yeah ignore this.